lib/reader/src/parser.rs

```diff
@@ -661,17 +676,21 @@ impl<'a> Parser<'a> {
     //
     // signature ::=  * "(" [arglist] ")" ["->" retlist] [call_conv]
     //
-    fn parse_signature(&mut self) -> Result<Signature> {
+    fn parse_signature(&mut self, unique_isa: Option<&TargetIsa>) -> Result<Signature> {
         let mut sig = Signature::new();
 
         self.match_token(Token::LPar, "expected function signature: ( args... )")?;
         // signature ::=  "(" * [arglist] ")" ["->" retlist] [call_conv]
         if self.token() != Some(Token::RPar) {
-            sig.argument_types = self.parse_argument_list()?;
+            sig.argument_types = self.parse_argument_list(unique_isa)?;
         }
         self.match_token(Token::RPar, "expected ')' after function arguments")?;
         if self.optional(Token::Arrow) {
-            sig.return_types = self.parse_argument_list()?;
+            sig.return_types = self.parse_argument_list(unique_isa)?;
+        }
+
+        if sig.argument_types.all(ArgumentLoc::is_assigned) {
```

Comment:
Oh dear, can't believe I forgot to compile a last time before commit, sorry about that